### ISSUE TITLE
promtool return zero code on validation error (unmarshall errors)

### ISSUE THREAD

**Issue by thrix** [prometheus/prometheus#10913]
### What did you do?

```
$ promtool check config inputs/prometheus/prometheus.yml; echo $?
  FAILED: parsing YAML file /tmp/prometheus.yml: yaml: unmarshal errors:
  line 22: field relabel_configs not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }
  line 23: field source_labels not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }
  line 24: field target_label not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }
  line 25: field source_labels not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }
  line 26: field target_label not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }
  line 27: field target_label not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }
  line 28: field replacement not found in type struct { Targets []string "yaml:\"targets\""; Labels model.LabelSet "yaml:\"labels\"" }

0
```

### What did you expect to see?

Promtool fails with non-zero return code

### What did you see instead? Under which circumstances?

Zero return code

### System information

Linux 5.17.7-300.fc36.x86_64 x86_64

### Prometheus version

```text
promtool, version 2.36.2 (branch: HEAD, revision: d7e7b8e04b5ecdc1dd153534ba376a622b72741b)
  build user:       root@f051ce0d6050
  build date:       20220620-13:21:35
  go version:       go1.18.3
  platform:         linux/amd64
```

### Prometheus configuration file

```yaml
global:
  scrape_interval:     20s
  evaluation_interval: 5s

# Alertmanager configuration
alerting:
  alertmanagers:
    - static_configs:
      - targets:
        - alertmanager:9093

# Scrape configurations
scrape_configs:
  - job_name: "invalid"
    metrics_path: /probe
    params:
      module:
        - http_fuse
    static_configs:
      - targets:
        - https://someurl
        relabel_configs:
      - source_labels: [__address__]
        target_label: __param_target
      - source_labels: [__param_target]
        target_label: instance
      - target_label: __address__
        replacement: blackbox-exporter:9115
```

**Comment by Equwece** [comment]
Hello, downgrading Prometheus to 2.36.0 solved this problem for me.

**Comment by roidelapluie** [comment]
Thanks for reporting this bug, it is solved in #10952
